clang/include/clang/AST/PreorderAST.h

```diff
@@ -198,6 +198,20 @@ namespace clang {
     // from outside this class and invokes Cleanup on the root node which
     // recursively deletes the AST.
     void Cleanup() { Cleanup(Root); }
+
+    bool operator<(PreorderAST &Other) {
```

Comment:
Making these operators `const` results in an error: `'clang::PreorderAST::Compare': 2 overloads have no legal conversion for 'this' pointer` since `PreorderAST::Compare` cannot be `const`.